cli/whoami.go

```diff
@@ -0,0 +1,38 @@
+package cli
+
+import (
+	"fmt"
+
+	"github.com/coder/coder/v2/cli/cliui"
+	"github.com/coder/coder/v2/codersdk"
+	"github.com/coder/pretty"
+	"github.com/coder/serpent"
+)
+
+func (r *RootCmd) whoami() *serpent.Command {
+	client := new(codersdk.Client)
+	cmd := &serpent.Command{
+		Annotations: workspaceCommand,
+		Use:         "whoami",
+		Short:       "Fetch authenticated user info for Coder deployment",
+		Middleware: serpent.Chain(
+			serpent.RequireNArgs(0),
+			r.InitClient(client),
+		),
+		Handler: func(inv *serpent.Invocation) error {
+			ctx := inv.Context()
+			// Fetch the user info
+			resp, err := client.User(ctx, codersdk.Me)
+			// Get Coder instance url
+			deployConfig, _ := client.DeploymentConfig(ctx)
```

Comment:
Not all users can query this endpoint. If we just want the url though, we can use `client.URL` instead.

Reply:
Ohh, Ok I didn't know that